src/ruby.ts

```diff
@@ -32,200 +46,199 @@ export class RubyProcess extends EventEmitter {
 	public constructor(args: LaunchRequestArguments) {
 		super();
 		this.launchArgs = args;
+		this.pendingResponses = [];
 		this.pendingCommands = [];
+		this.socketConnected = false;
 		this.state = SocketClientState.ready;
 
-		var that = this;
 		var runtimeArgs = ['--evaluation-timeout', '10'];
 		var runtimeExecutable: string;
 
 		if (process.platform === 'win32') {
 			runtimeExecutable = 'rdebug-ide.bat';
 		}
-		else if (process.platform === 'darwin') {
-			runtimeExecutable = 'rdebug-ide';
-		}
 		else {
 			// platform: linux
 			runtimeExecutable = 'rdebug-ide';
 		}
 
-		var programArgs = [];
 		var processCwd = dirname(this.launchArgs.program);
+		//xd only if showDebuggerOutput is set - it is also a runtime arg, not a program arg
```

Comment:
I consider the runtime to be the debugger, since we aren't passing arguments directly to ruby. Note that the `d` flag (line tracing) doesn't work anyway.

Added some explanation around this.